deploy/util/filelineswap.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+# imports needed for python type checking
+from typing import List
+
+def file_line_swap(in_file: str, out_file: str, start_marker: str, end_marker: str, inject_lines: List[str]):
+    """Inject a set of lines into a file given two markers. Output into a new file.
+
+    Ex.
+        In:
+            Temp
+            # in mark
+            Temp2
+            # out mark
+            Temp3
+
+        Run w/ start_marker = "in" and out_marker = "out" and inject_lines = [Hi\n, I'm, Jeremy]
+
+        Out:
+            Temp
+            Hi
+            I'mJeremy
+            Temp3
+
+    Args:
+        in_file: input file to use as a base
+        out_file: output file to write into
+        start_marker: use this to determine start of replace area
+        end_marker: use this to determine end of replace area
+        inject_lines: lines to inject in the replacement area
+    """
+    with open(in_file, "r") as f:
+        og_lines = f.readlines()
+
+    with open(out_file, "w") as f:
```

Comment:
Can you make this a little bit stricter and ensure the existence of exactly one start and end marker string.

Reply:
I meant without parsing (just tracking indent/path to root as you go through the lines). You can just leave this as is and I can update it in the metasim PR.